Cython/Coverage.py

```diff
@@ -50,13 +50,22 @@ def _find_dep_file_path(main_file, file_path, relative_path_search=False):
     return canonical_filename(abs_path)
 
 
+class NoCoverConfigurer(CoveragePlugin):
+    exclude_lines = []
+
+    def configure(self, config):
+        type(self).exclude_lines = config.get_option("report:exclude_lines")
```

Comment:
Is it correct to use `type(self)…` here? This feels too global for a module specific result. Doesn't this work as well?
```suggestion
        # Pass on the "excluded_lines" as found by the "CythonModuleReporter" below.
        self.exclude_lines = config.get_option("report:exclude_lines")
```

Reply:
I am not fond of global variables either. However, I could not find a method on `Plugin` that would receive `Configurer` instances so that I do not think that an instance attribute would work.

Reply:
That's pretty much why I was asking, below you have:
> reg.add_configurer(NoCoverConfigurer())`
so it _is_ passing an instance. I'll commit this and see what it gives.

Reply:
Looks like it's not the same thing:
```
  File "collect_coverage.py", line 90, in run_json_report
    assert summary['missing_lines'] == 2, summary
AssertionError: {'covered_lines': 8, 'num_statements': 12, 'percent_covered': 66.66666666666667, 'missing_lines': 4, 'excluded_lines': 0}
```

Reply:
No, I do not think it is. The calls in `coverage_init()` register the classes with the `coverage` library but I am not sure that users are supposed to keep track of the instances. The globals are ugly but coverage runs once and exits.

Reply:
> coverage runs once and exits

If you run it from the CLI, then yes. There's an API interface to it, too, though, and I'd like to keep that working. I'd say that prohibits global state.

Reply:
Yes, I understand but the configuration is fairly static: Either a file on the filesystem or the global defaults.

There may be a more elegant way to achieve this, however. I could not find any fitting example and the documentation only describes the API.